### PR TITLE
Adding kvasir mpl

### PR DESCRIPTION
**Pull request by odinthenerd**
This PR adds the kvasir mpl to metabench. Kvasir mpl is implemented using the meta continuations pattern which leverages alias calls more than other strategies in order to speed up compile time. It is therefore probably interesting here although it is not (yet) widely used.

### CODE REVIEW

File: benchmark/dependencies/kvasir.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    AND NOT (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+             CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.7") 
+    AND NOT (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND
+             CMAKE_CXX_COMPILER_VERSION VERSION_LESS "19")
+)
+#    find_package(Kvasir QUIET) <== Insert your logic to find preinstalled Kvasir
+#    if (Kvasir_FOUND)
+#        message(STATUS "Local Kvasir installation found - version ${Kvasir_VERSION}")
+#        add_custom_target(Kvasir)
+#    else()
```
Comment (brunocodutra) [inline benchmark/dependencies/kvasir.cmake]: Please remove this commented out section

**Comment by odinthenerd** [comment]
damn sorry, we need to fix something on our side, ignore this for a day or two

File: benchmark/type/all/CMakeLists.txt [hunk 2]
```diff
@@ -5,6 +5,7 @@
 set(datasets)
 add_dataset(datasets type all brigand list brigand.list.cpp.erb 15 "(0..50).step(10).to_a + (100..500).step(50).to_a")
 add_dataset(datasets type all metal   list metal.list.cpp.erb   15 "(0..50).step(10).to_a + (100..500).step(50).to_a")
+add_dataset(datasets type all kvasir   list kvasir.list.cpp.erb   15 "(0..50).step(10).to_a + (100..500).step(50).to_a")
```
Comment (brunocodutra) [inline benchmark/type/all/CMakeLists.txt]: could you align the columns and keep it sorted alphabetically?

File: benchmark/type/any/CMakeLists.txt [hunk 3]
```diff
@@ -6,5 +6,6 @@ set(datasets)
 add_dataset(datasets type any brigand list brigand.list.cpp.erb 15 "(0..50).step(10).to_a + (100..500).step(50).to_a")
 add_dataset(datasets type any meta    list meta.list.cpp.erb    15 "(0..50).step(10).to_a + (100..500).step(50).to_a")
 add_dataset(datasets type any metal   list metal.list.cpp.erb   15 "(0..50).step(10).to_a + (100..500).step(50).to_a")
+add_dataset(datasets type any kvasir   list kvasir.list.cpp.erb   15 "(0..50).step(10).to_a + (100..500).step(50).to_a")
```
Comment (brunocodutra) [inline benchmark/type/any/CMakeLists.txt]: same here

File: benchmark/type/partition/CMakeLists.txt [hunk 4]
```diff
@@ -6,6 +6,7 @@ set(datasets)
 add_dataset(datasets type partition brigand list   brigand.list.cpp.erb 10 "(0..50).step(10).to_a + (100..500).step(50).to_a")
 add_dataset(datasets type partition meta    list   meta.list.cpp.erb    10 "(0..50).step(10).to_a + (100..500).step(50).to_a")
 add_dataset(datasets type partition metal   list   metal.list.cpp.erb   10 "(0..50).step(10).to_a + (100..500).step(50).to_a")
+add_dataset(datasets type partition kvasir  list   kvasir.list.cpp.erb  10 "(0..50).step(10).to_a + (100..500).step(50).to_a")
```
Comment (brunocodutra) [inline benchmark/type/partition/CMakeLists.txt]: please sort alphabetically

File: benchmark/type/filter/CMakeLists.txt [hunk 5]
```diff
@@ -6,6 +6,7 @@ set(datasets)
 add_dataset(datasets type filter brigand list   brigand.list.cpp.erb 10 "(0..50).step(10).to_a + (100..500).step(50).to_a")
 add_dataset(datasets type filter meta    list   meta.list.cpp.erb    10 "(0..50).step(10).to_a + (100..500).step(50).to_a")
 add_dataset(datasets type filter metal   list   metal.list.cpp.erb   10 "(0..50).step(10).to_a + (100..500).step(50).to_a")
+add_dataset(datasets type filter kvasir  list   kvasir.list.cpp.erb  10 "(0..50).step(10).to_a + (100..500).step(50).to_a")
```
Comment (brunocodutra) [inline benchmark/type/filter/CMakeLists.txt]: please sort alphabetically

File: benchmark/CMakeLists.txt [hunk 6]
```diff
@@ -43,6 +45,8 @@ macro(add_dataset datasets category case lib datatype path_to_template k range)
         Meta_add_dataset(${args})
     elseif("${lib}" STREQUAL "metal")
         Metal_add_dataset(${args})
+    elseif("${lib}" STREQUAL "kvasir")
+        Kvasir_add_dataset(${args})
```
Comment (brunocodutra) [inline benchmark/CMakeLists.txt]: please sort alphabetically

**Comment by brunocodutra** [comment]
Looks good

**Comment by odinthenerd** [comment]
squashed into new branch, this can be removed

**Pull request closed without merging**